File: training-data/php/quicksort.php
<?php

function quicksort(array $items): array
{
    if (count($items) <= 1) {
        return $items;
    }

    $pivot = $items[intdiv(count($items), 2)];
    $left = array_values(array_filter($items, fn($x) => $x < $pivot));
    $mid = array_values(array_filter($items, fn($x) => $x === $pivot));
    $right = array_values(array_filter($items, fn($x) => $x > $pivot));

    return array_merge(quicksort($left), $mid, quicksort($right));
}

print_r(quicksort([5, 3, 8, 1, 9, 2]));

File: training-data/swift/quicksort.swift
func quicksort(_ items: [Int]) -> [Int] {
    if items.count <= 1 { return items }
    let pivot = items[items.count / 2]
    let left = items.filter { $0 < pivot }
    let mid = items.filter { $0 == pivot }
    let right = items.filter { $0 > pivot }
    return quicksort(left) + mid + quicksort(right)
}

print(quicksort([5, 3, 8, 1, 9, 2]))
